core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java

```diff
@@ -93,6 +93,15 @@ void modifyProperties(Consumer<Map<String,String>> mapMutator) throws AccumuloEx
    */
   Map<String,String> getSystemConfiguration() throws AccumuloException, AccumuloSecurityException;
 
+  /**
+   * Retrieves the stored System properties from zookeeper. This method is different from
+   * {@link #getSystemConfiguration()} as it will only return the stored properties in zookeeper and
+   * not properties from accumulo.properties or default values..
+   *
+   * @return a map of stored system properties set in zookeeper
+   */
+  Map<String,String> getStoredProperties() throws AccumuloException, AccumuloSecurityException;
```

Comment:
Again, the word `stored` here does not convey how these properties are different than the ones returned by `getSystemConfiguration`. The user will need to look at the documentation to know, so IMO we should leave the name  as `getSystemProperties`.

Reply:
Sounds good, we can go back to the old name. I would only want to change it if there is agreement which there is not.

Reply:
What about `getOverridenProperties`? This would coincide with the "overridden" label displayed by the Shell config command.

Reply:
Stored or something equivalent seems to point in the correct direction - they would need to look at the documentation to find out what the difference between `getSystemConfiguration` and `getSystemProperies` difference is anyway - not having it named properties in the code (too me) increases clarity because properties and configuration seem to be easily confused

Reply:
I dunno, the term "override" could be confusing too because in the shell we also use that for site config (accumulo.properties) and ZK properties. For example, if I set "tserver.wal.max.size", the shell displays:
<pre>
default    | tserver.wal.max.size ............................................. | 1G
site       |    @override ..................................................... | 512M
system     |    @override ..................................................... | 400M
</pre>

Reply:
So, I looked at NamespaceOperations and TableOperations. Both of them have a `getProperties` method that returns properties scoped at the appropriate level. On InstanceOperations we have:

getSystemConfiguration() -> returns the entire running configuration
getSiteConfiguration() -> returns the properties set in `accumulo.properties`
getSystemProperties() -> we're proposing to resurrect this method because of a failed test, which would return only the properties stored in ZooKeeper.

Why would a user, via the AccumuloClient, need to know where the properties are stored? Why should they care? I can see this is a server-side capability, but why does it exist in the client API? I would propose the following:

  1. that we modify InstanceOperations to have a `getProperties` method that returns properties at the instance level just like NamespaceOperations and TableOperations.
  2. That we deprecate (and remove) getSiteConfiguration and getSystemProperties in InstanceOperations. There is no reason that PropStoreConfigIT can't use a server-side API to get the information that it needs for the test.

Reply:
For instance operations, `getSiteConfiguration` seems to still makes sense to fetch what is set via the configuration.

Reply:
Why would the user need to know what is set in `accumulo.properties` vs what is stored in `ZooKeeper`. `InstanceOperations.getSiteConfiguration` connects to a random tablet server to get the configuration on that server. The `accumulo.properties` file on that server may be different than other servers.

Reply:
1) If I have shell access that is different from needing access to read files on a system.
2) There is a hash check that checks properties have the same critical values so there is a good probability that an active tserver has properties that match across the cluster.

Reply:
> If I have shell access that is different from needing access to read files on a system.
> There is a hash check that checks properties have the same critical values so there is a good probability that an active tserver has properties that match across the cluster.

True statements, but you didn't address the why. I still don't understand the utility here.

Reply:
Based on the discussion not being settled for now I think it's best to just fix the test so I pushed a commit to do that. We could create another follow on Issue to discuss about changing the API and what methods should be exposed on InstanceOperations if we want to do that.